.github/workflows/build-and-test.yml

```diff
@@ -16,14 +15,31 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Checkout 
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
       timeout-minutes: 1
     - name: Touch .env  
       run: touch .env
       timeout-minutes: 1
     - name: Prepare
-      run: docker compose build test
+      run: echo lol # docker compose build test
       timeout-minutes: 1
     - name: Test
-      run: docker compose run test
+      run: echo whut # docker compose run test
       timeout-minutes: 1
+
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    needs: build
+    if: ${{ github.event_name == 'push' && contains(fromJSON('["refs/heads/master", "refs/heads/module-9-exercise"]'), github.ref) }}
+    steps:
+    - name: Login to Docker Hub
+      run: echo ${{ secrets.DOCKERHUB_KEY }} | docker login -u sweavo --password-stdin
+      timeout-minutes: 1
+    - name: Checkout 
+      uses: actions/checkout@v4
+      timeout-minutes: 1
+    - name: Deploy via makefile
+      run: make deploy-docker && curl -dH -X POST '${{ secrets.WEBAPP_RESTART_WEBHOOK }}'
```

Comment:
One nice addition here can be adding the `--fail` or [`--fail-with-body` arguments](https://curl.se/docs/manpage.html#--fail-with-body) to curl here, since otherwise curl will report complete even if Azure sends back a 4xx or 5xx error which can easily lead to failed pipelines being missed